tests/e2e/gc_rwx_syncer.go

```diff
@@ -0,0 +1,866 @@
+
+		ginkgo.By(fmt.Sprintf("Waiting for labels %+v to be updated for pv %s", pvLabels1, pv1.Name))
+		err = e2eVSphere.waitForLabelsToBeUpdated(fcdIDInCNS,
+			pvLabels1, string(cnstypes.CnsKubernetesEntityTypePV), pv1.Name, pv1.Namespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		pvLabels2 := make(map[string]string)
+		pvLabels2["pvLabelKeySyncer2"] = "pv-label-Value2"
+
+		ginkgo.By(fmt.Sprintf("Updating labels %+v for pv %s in namespace %s", pvLabels2, pv2.Name, namespace))
+		pv2, err = client.CoreV1().PersistentVolumes().Get(ctx, pv2.GetName(), metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		pv2.Labels = pvLabels2
+
+		pv2, err = client.CoreV1().PersistentVolumes().Update(ctx, pv2, metav1.UpdateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By(fmt.Sprintf("Waiting for labels %+v to be updated for pv %s", pvLabels2, pv2.Name))
+		err = e2eVSphere.waitForLabelsToBeUpdated(fcdIDInCNS,
+			pvLabels2, string(cnstypes.CnsKubernetesEntityTypePV), pv2.Name, pv2.Namespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Create a Pod to use this PVC
+		ginkgo.By("Creating pod to attach PV1 to the node")
+		pod, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, execRWXCommandPod1)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			// Delete POD
+			ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, namespace))
+			err = fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Verifying whether the CnsFileAccessConfig CRD is Deleted or not for Pod")
+			verifyCNSFileAccessConfigCRDInSupervisor(ctx, f, pod.Spec.NodeName+"-"+pvcNameInSV,
+				crdCNSFileAccessConfig, crdVersion, crdGroup, false)
+		}()
+
+		// Create a Pod to use this PVC
+		ginkgo.By("Creating pod2 to attach PV2 to the node")
+		pod2, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvc2}, false, execRWXCommandPod2)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			// Delete POD
+			ginkgo.By(fmt.Sprintf("Deleting the pod2 %s in namespace %s", pod2.Name, namespace))
+			err = fpod.DeletePodWithWait(client, pod2)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Verifying whether the CnsFileAccessConfig CRD is Deleted or not for Pod2")
+			verifyCNSFileAccessConfigCRDInSupervisor(ctx, f, pod2.Spec.NodeName+"-"+pvcNameInSV,
+				crdCNSFileAccessConfig, crdVersion, crdGroup, false)
+		}()
+
+		gcClusterID := strings.Replace(pvcNameInSV, pvcUID, "", -1)
+		framework.Logf("gcClusterId " + gcClusterID)
+
+		pv1UID := string(persistentvolumes[0].UID)
+		framework.Logf("PV1 uuid " + pv1UID)
+
+		pv2UID := string(pv2.UID)
+		framework.Logf("PV2 uuid " + pv2UID)
+
+		podUID := string(pod.UID)
+		framework.Logf("Pod uuid : " + podUID)
+
+		//Add a check to validate CnsVolumeMetadata crd
```

Comment:
we should validate for both GC PVCs/PVs right

Reply:
Added validation check for both pvc and pv